DSCResources/MSFT_xDefaultGatewayAddress/MSFT_xDefaultGatewayAddress.psm1

```diff
@@ -0,0 +1,247 @@
+######################################################################################
+function Get-TargetResource
+{
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (        
+        [String]$Address,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet("IPv4", "IPv6")]
+        [String]$AddressFamily
+    )
+    
+    # Use $AddressFamily to select the IPv4 or IPv6 destination prefix
+    $DestinationPrefix = "0.0.0.0/0"
+    if($AddressFamily -eq "IPv6")
+    {
+        $DestinationPrefix = "::/0"
+    }
+    # Get all the default routes
+    $defaultRoutes = Get-NetRoute -InterfaceAlias $InterfaceAlias -AddressFamily `
+        $AddressFamily -ErrorAction Stop | `
+        where { $_.DestinationPrefix -eq $DestinationPrefix }
+
+    $returnValue = @{
+        AddressFamily = $AddressFamily
+        InterfaceAlias = $InterfaceAlias
+    }
+    # If there is a Default Gateway defined for this interface/address family add it
+    # to the return value.
+    if ($defaultRoutes) {
+        $returnValue += @{ Address = $DefaultRoutes.NextHop }
+    }
+
+    $returnValue
+}
+
+######################################################################################
+# The Set-TargetResource cmdlet.
+# This function will set the Default Gateway Address for the Interface/Family in the
+# current node
+######################################################################################
+function Set-TargetResource
+{
+    param
+    (    
+        [String]$Address,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet("IPv4", "IPv6")]
+        [String]$AddressFamily
+    )
+
+    ValidateProperties @PSBoundParameters -Apply
+}
+
+######################################################################################
+# The Test-TargetResource cmdlet.
+# This will test if the given Address is set as the Gateway Server address for the
+# Interface/Family in the current node
+######################################################################################
+function Test-TargetResource
+{
+    [OutputType([System.Boolean])]
+    param
+    (        
+        [String]$Address,
+
+        [Parameter(Mandatory)]
+        [ValidateNotNullOrEmpty()]
+        [String]$InterfaceAlias,
+
+        [Parameter(Mandatory)]
+        [ValidateSet("IPv4", "IPv6")]
+        [String]$AddressFamily
+    )
+
+    ValidateProperties @PSBoundParameters
+}
+
+
+#######################################################################################
+#  Helper function that validates the Gateway Address property. If the switch parameter
+# "Apply" is set, then it will set the properties after a test
+#######################################################################################
+function ValidateProperties
```

Comment:
Please use the verb-noun function name here. Suggesiton: Test-GatewayProperties, or just Test-Properties.